### ISSUE TITLE
PaymentManager behavior unclear

### ISSUE THREAD

**Issue by domenic** [w3c/webpayments-payment-apps-api#253]
What happens if I do something like the following?

```js
const registration = await navigator.serviceWorker.register('/sw.js');

setInterval(() => {
  registration.paymentManager.userHint = Math.random().toString();
}, 1000);

setInterval(() => {
  registration.paymentManager.instruments.set(..., ...);
}, 1500);
```

That is, right now there are no restrictions on when the instruments collection and the user hints can be update. Note that since ServiceWorkerRegistration is exposed both inside and outside the service worker, the service worker itself can update these, e.g. in response to push notifications or fetch events.

Is that desired and expected? It seemed like from reading the document, the expectation was that these things would be set once, at registration time.

**Comment by rsolomakhin** [comment]
Setting instruments must happen after registration and is required to become a payment handler. That explains the code structure in the examples. However, we do expect that service workers will occasionally update the instruments in background. For example, suppose you open up `https://my-bank.com` on your laptop and activate a new credit card. Then the service worker on your phone should add this new instrument to the PaymentManager. This would happen in background, without the user having to navigate to `https://my-bank.com` on their phone.

**Comment by jmpas** [comment]
Also confused by this, while working on a proof of concept with "just in time" installation, I tried setting up instruments from inside the service worker `self.registration.paymentManager.instruments.set(..., ...)`, specifically in the `paymentrequest` event callback, but the browser tab crashes on chrome when doing so (couldn't find any meaningful error from it when debugging).

It's not clear to me `when` and `where` it's possible to set instruments besides the common example...

Maybe it would be nice to have a list of use cases ⁄ scenarios on the spec 🤷‍♂️

**Comment by rsolomakhin** [comment]
@Nipher : Could you please click on "Provide additional details" in chrome://crashes/ ? This will file a bug report on crbug.com. Please send me this link either here or to rouslan@chromium.org. Thank you!

**Comment by rsolomakhin** [comment]
@Nipher: Your use case should work, btw.

**Comment by ianbjacobs** [comment]
I agree we should ensure that the spec makes clear that instruments may be updated after registration. (I am happy to work with @rsolomakhin on text.)

**Comment by jmpas** [comment]
@rsolomakhin I've tested it on chromium today and it works
Looks like a chrome specific problem then? should I still file the report and send the link to you?

Thanks! 🙇

**Comment by rsolomakhin** [comment]
Yes, please report it for Chrome. @Nipher .
